addons/gamification/models/res_users.py

```diff
@@ -42,35 +64,29 @@ def _get_user_badge_level(self):
     @api.model_create_multi
     def create(self, values_list):
         res = super(Users, self).create(values_list)
-
-        karma_trackings = []
-        for user in res:
-            if user.karma:
-                karma_trackings.append({'user_id': user.id, 'old_value': 0, 'new_value': user.karma})
+        tracking = self.env['gamification.karma.tracking']
+
+        karma_trackings = [
+            tracking.create_tracking(
+                user.id, 0, vals.get('karma', 0), reason=_('User Creation'))
+            for user, vals in zip(res, values_list)
+            if vals.get('karma', 0)
```

Comment:
`.get` return None, so no need of default value for the condition
```suggestion
            if vals.get('karma')
```

Reply:
done